File: Python/594. Longest Harmonious Subsequence.py
class Solution:
    def findLHS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = collections.Counter(nums)
        ans = 0
        for num in nums:
            if count[num] > 0 and count[num+1] > 0:
                ans = max(ans, count[num]+count[num+1])
        return ans

File: C++/594. Longest Harmonious Subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int, int>maple;
        for(auto num: nums)
            maple[num] += 1;
        int ans = 0;
        for(auto num: nums)
        {
            if(maple[num+1]>0)
                ans = max(ans,maple[num]+maple[num+1]);
        }
        return ans;
    }
};